client-js/app/elements/expandable-input.html

```diff
@@ -0,0 +1,97 @@
+<!-- 
+Expandable input. Takes a registry key value and shortens it to fit on the 
+screen. When clicked on reveals an editable text box which is shifted into 
+focus. Enter fires 'iron-form-submit' which sends the key, value pair as 
+event.detail
+ -->
+
+<dom-module id="expandable-input">
+  <style>
+  #long{
+    visibility: hidden;
+    position: absolute;
+    width: 80%;
+  }
+  </style>
+  <template >
+    <iron-a11y-keys keys="enter" on-keys-pressed="submitForm">
+    <!-- TODO remap newline from enter to "ctrl enter" -->
+      <div id="long">
+        <textarea id="input" value="{{value::input}}" rows="[[compRows]]" cols="[[cols]]"></textarea>
+      </div>
+      <div id="short">
+        <paper-item on-click="selectField" id="static" label="{{valueShort}}">{{valueShort}}</paper-item>
+      </div>
+    </iron-a11y-keys>
+  </template>
+  <script>
+  Polymer({
```

Comment:
I don't know typescript. My plan was to write this in js and convert it to ts as an excercise for myself.